lib/action_subscriber/middleware/env.rb

```diff
@@ -38,7 +38,7 @@ def initialize(subscriber, encoded_payload, properties)
         @has_been_nacked = false
         @has_been_rejected = false
         @headers = properties.fetch(:headers, {})
-        @message_id = properties.fetch(:message_id, ::SecureRandom.hex(3))
+        @message_id = properties.try(:message_id) ? properties.fetch(:message_id) : ::SecureRandom.hex(3)
```

Comment:
You can use the new(ish) safe operator here and avoid the penalty of using `try`:

```ruby
@message_id = properties.&message_id ? properties.fetch(:message_id) : ::SecureRandom.hex(3)
```

However, I don't see how this will prevent us from capturing an empty string. Perhaps we should be [checking it's presence](https://api.rubyonrails.org/classes/Object.html#method-i-presence) instead?

```ruby
@message_id = properties.message_id.presence || ::SecureRandom.hex(3)
```

Reply:
:+1: for the `presence` check. I think the reason we're not receiving a message id is because we're not setting one on the publisher side.

Reply:
@film42 That's a good point.  MessageID is an application use thing in AMQP.  I was thinking of how we might be able to make this useful.  We might want to keep this here, but I was thinking for logging purposes it might be more useful to have something like `channelID - deliveryTag`